core/modules/search/search.module

```diff
@@ -111,13 +111,98 @@ function search_block_info() {
   return $blocks;
 }
 
+/**
+ * Implements hook_block_configure().
+ */
+function search_block_configure($delta = '', $settings = array()) {
+  $form = array();
+  if ($delta == 'form') {
+    if (array_key_exists('search', $settings) && !empty($settings['search'])) {
```

Comment:
Isn't this the same as `if (!empty($settings['search'])) {`?

Reply:
depends on the php version? /me goes to look that up...

Reply:
Youd still need to declare `$search_settings = array()` before this. The PHP error persists.

Reply:
yes. And it also looks like `array_key_exists()` is a lot slower than `isset()`! learning things :)

Reply:
> The PHP error persists.

What are your steps to reproduce the error? I'm not getting it...  I thought the `$search_settings += ` should handle the assignment.

Reply:
Nevermind, I got it! pushing a fix now :)